### ISSUE TITLE
trackpad is intermittent with driver installed

### ISSUE THREAD

**Issue by Kylar514** [amiablechief/Magic-Trackpad-2024-USB-C-Linux#1]
I just switched to the kernal you are using, 6.12.1 zen.  Had the same issue on arch-arch kernal.  

symptoms:
- no input unless holding down a click, or rapidly moving finger across pad like a maniac, at which point it is not usable.
- 2 finger scrolling works intermitently, spamming 2 fingers up and down will move the mouse up and and down, but if i go fast enough i might get a scroll in 1 direction or the other, also unusable.
- right click, have no idea how that happens, it just does when im spamming inputs lol, i've gotten it to proc like 3x.
- zoom works surprisingly well, not perfect, but its probably the most reliable funnily enough.

I do use hyprland and libinput.  Not really sure whats causing it to act like this.  uninstalling or disabling the driver will allow me to move the mouse around on the trackpad flawlessly, I am just limited to a single left click, no scroll, no right click, etc.

**Comment by amiablechief** [comment]
This sounds like the trackpad surface might be unclean? Could you try cleaning it with microfiber or similar and see if that helps? The only other thing that's different is I'm using KDE and have no experience with hyprland though so not sure how much of this is hyprland acting up (though logically it shouldn't). Have you tried using `libinput debug-events` to see what outputs are appearing as you touch/click, etc?

**Comment by Kylar514** [comment]
yeap its clean, brand new out of the box.

I been using `evtest` but just tried `libinput debug-events`, and well evtest gives ALOT of events, with every touch.  even if im not moving my finger.  and libinput is substantially less data, but the data that output matches the erraticness of the behavior.  so if i do manage to proc a 2 finger gesture scroll, it does display it, but when its not procing, i dont get data, or sometimes it'll register as a pointer_motion.  etc.  Here is a gif of the dif.

[gif imgur, left side evtest, right side libinput debug.  i just piped libinput into grep event3, these should have the same output but clearly they do not.](https://imgur.com/Fe4Cikq)

So the weird thing is if I dont install this driver, or uninstall it, I can use the trackpad just fine.  I am limited to moving the cursor and a single hard press, and it seems to be using the hid-generic driver.  I tried disalbing that driver, (it disables my keyboard though) and it will stop the trackpad from working when this driver is not installed, and has no effect when this driver is installed.

I was certain that it was a conflicting driver issue but cant really be certain anymore..  At this point i did disable secure boot and reinstalled.

That said, when you ran the install script, did you get any depracation errors when installing?  Here is what I get from a fresh install, followed up with dkms status.


``` gitclones 󱍿 pacman -Ss dkms # or paru, yay, yum, etc
git clone https://github.com/amiablechief/Magic-Trackpad-2024-USB-C-Linux.git
cd Magic-Trackpad-2024-USB-C-Linux
chmod u+x install.sh
sudo ./install.sh
extra/acpi_call-dkms 1.2.2-2
    A linux kernel module that enables calls to ACPI methods through /proc/acpi/call - module sources
extra/bbswitch-dkms 0.8-713
    Kernel module allowing to switch dedicated graphics card on Optimus laptops
extra/broadcom-wl-dkms 6.30.223.271-38
    Broadcom 802.11 Linux STA wireless driver
extra/deepin-anything-dkms 5.0.18-1
    Deepin Anything file search tool
extra/dkms 3.1.3-1 [installed]
    Dynamic Kernel Modules System
extra/ndiswrapper-dkms 1.62-5
    Module for NDIS (Windows Network Drivers) drivers supplied by vendors
extra/nvidia-dkms 565.57.01-2
    NVIDIA kernel modules - module sources
extra/nvidia-open-dkms 565.57.01-2
    NVIDIA open kernel modules - module sources
extra/openrazer-driver-dkms 3.9.0-1
    OpenRazer kernel modules sources
extra/sysdig-dkms 0.39.0-1
    DKMS kernel module for sysdig
extra/v4l2loopback-dkms 0.13.2-1 [installed]
    v4l2-loopback device – module sources
extra/vhba-module-dkms 20240917-14
    Kernel module that emulates SCSI devices
extra/virtualbox-host-dkms 7.1.4-2
    VirtualBox Host kernel modules sources
Cloning into 'Magic-Trackpad-2024-USB-C-Linux'...
remote: Enumerating objects: 598, done.
remote: Counting objects: 100% (598/598), done.
remote: Compressing objects: 100% (240/240), done.
remote: Total 598 (delta 214), reused 598 (delta 214), pack-reused 0 (from 0)
Receiving objects: 100% (598/598), 152.48 KiB | 2.09 MiB/s, done.
Resolving deltas: 100% (214/214), done.
[sudo] password for zeal:
+++ dirname ./install.sh
++ cd .
++ pwd
+ DIR=/home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux
+ chmod u+x /home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/scripts/install.sh
+ /home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/scripts/install.sh
+ dkms_name=hid-magicmouse-dkms
+ dkms_version=6.12.1-zen1-1-zen+magicmouse2
+++ dirname /home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/scripts/install.sh
++ cd /home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/scripts
++ pwd
+ DIR=/home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/scripts
+ dkms status -m hid-magicmouse-dkms -v 6.12.1-zen1-1-zen+magicmouse2
+ egrep '(added|built|installed)'
egrep: warning: egrep is obsolescent; using grep -E
+ dkms add /home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/scripts/../linux/drivers/hid
Deprecated feature: REMAKE_INITRD (/home/zeal/gitclones/Magic-Trackpad-2024-USB-C-Linux/linux/drivers/hid/dkms.conf)
Deprecated feature: REMAKE_INITRD (/usr/src/hid-magicmouse-dkms-6.12.1+magicmouse2/dkms.conf)
Creating symlink /var/lib/dkms/hid-magicmouse-dkms/6.12.1+magicmouse2/source -> /usr/src/hid-magicmouse-dkms-6.12.1+magicmouse2
+ dkms status -m hid-magicmouse-dkms -v 6.12.1-zen1-1-zen+magicmouse2
+ egrep '(built|installed)'
egrep: warning: egrep is obsolescent; using grep -E
+ dkms build hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2

Sign command: /usr/lib/modules/6.12.1-zen1-1-zen/build/scripts/sign-file
Signing key: /var/lib/dkms/mok.key
Public certificate (MOK): /var/lib/dkms/mok.pub
Deprecated feature: REMAKE_INITRD (/usr/src/hid-magicmouse-dkms-6.12.1-zen1-1-zen+magicmouse2/dkms.conf)
Creating symlink /var/lib/dkms/hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2/source -> /usr/src/hid-magicmouse-dkms-6.12.1-zen1-1-zen+magicmouse2

Cleaning build area... done.
Building module(s)... done.
Signing module /var/lib/dkms/hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2/build/hid-magicmouse.ko
Cleaning build area... done.
+ dkms status -m hid-magicmouse-dkms -v 6.12.1-zen1-1-zen+magicmouse2
+ egrep '(installed)'
egrep: warning: egrep is obsolescent; using grep -E
Deprecated feature: REMAKE_INITRD (/var/lib/dkms/hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2/source/dkms.conf)
+ dkms install --force hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2
Deprecated feature: REMAKE_INITRD (/var/lib/dkms/hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2/source/dkms.conf)

Found pre-existing /usr/lib/modules/6.12.1-zen1-1-zen/kernel/drivers/hid/hid-magicmouse.ko.zst, archiving for uninstallation
Installing /usr/lib/modules/6.12.1-zen1-1-zen/updates/dkms/hid-magicmouse.ko.zst
Running depmod... done.
+ sudo modprobe -a hid_magicmouse

```
```
(Main) Magic-Trackpad-2024-USB-C-Linux 󱍿 dkms status
Deprecated feature: REMAKE_INITRD (/var/lib/dkms/hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2/source/dkms.conf)
hid-magicmouse-dkms/6.12.1+magicmouse2: added
hid-magicmouse-dkms/6.12.1-zen1-1-zen+magicmouse2, 6.12.1-zen1-1-zen, x86_64: installed (Original modules exist)
v4l2loopback/0.13.2, 6.12.1-zen1-1-zen, x86_64: installed
```

**Comment by Kylar514** [comment]
ok.... lol.  ok.  So I fixed it haha...  Resolution is .... 

Prior to finding this repo, I tried something. setting up a quirks file to dictate how libinput would handle the magic trackpad.  Needless to say it didnt work.  Well until I installed this driver, at which point, it conflicted and caused all of my issues.  deleting the file fixed it.  Whats worse is during my troubleshooting process, I swear I deleted that file, but apparently I didn't.. so.  Once I deleted the file, all worked perfectly fine....

I also installed kde plasma, which I think was completely irrelavant to fixing it.. I plan to uninstall it now. 

Anyways thanks for your response, and the updated driver!
